TypeScript: Remove brackets with return+`as` statements+ternary operator

**Prettier 1.7.4**
**TypeScript 2.5.3**
[Playground link](https://prettier.io/playground/#%7B%22content%22%3A%22if%20(_.isBoolean(aValue)%20%26%26%20_.isBoolean(bValue))%20%7B%5Cn%20%20if%20(aValue%20as%20boolean)%20%7B%5Cn%20%20%20%20return%20(bValue%20as%20boolean)%20%3F%200%20%3A%20-1%3B%5Cn%20%20%7D%20else%20%7B%5Cn%20%20%20%20return%20(bValue%20as%20boolean)%20%3F%201%20%3A%200%3B%5Cn%20%20%7D%5Cn%7D%22%2C%22options%22%3A%7B%22ast%22%3Afalse%2C%22bracketSpacing%22%3Atrue%2C%22doc%22%3Afalse%2C%22jsxBracketSameLine%22%3Afalse%2C%22output2%22%3Afalse%2C%22parser%22%3A%22typescript%22%2C%22printWidth%22%3A80%2C%22semi%22%3Atrue%2C%22singleQuote%22%3Atrue%2C%22tabWidth%22%3A2%2C%22trailingComma%22%3A%22es5%22%2C%22useTabs%22%3Afalse%7D%7D)
```sh
--parser typescript
--single-quote
--trailing-comma es5
```

**Input:**
```tsx
if (_.isBoolean(aValue) && _.isBoolean(bValue)) {
  if (aValue as boolean) {
    return (bValue as boolean) ? 0 : -1;
  } else {
    return (bValue as boolean) ? 1 : 0;
  }
}
```

**Output:**
```tsx
if (_.isBoolean(aValue) && _.isBoolean(bValue)) {
  if (aValue as boolean) {
    return bValue as boolean ? 0 : -1;
  } else {
    return bValue as boolean ? 1 : 0;
  }
}

```

**Expected behavior:**
Don't remove brackets.
I get typescript error: `error TS1005: ';' expected.` 
I get this error both when I just start typescript, and when I restart prettier on the code already fixed by it:

![image](https://user-images.githubusercontent.com/7579698/31707468-dc6e1ce6-b405-11e7-85c8-467243c6f605.png)

I understand that this code can be rewritten (I already did it), this is already a question for me.
But I think that your tool should not break code, after using prettier the code should still work (compile), and the code behavior should remain the same.

--- a/src/fast-path.js
+++ b/src/fast-path.js
@@ -291,6 +291,9 @@ FastPath.prototype.needsParens = function(options) {
     case "TSAsExpression":
     case "LogicalExpression":
       switch (parent.type) {
+        case "ConditionalExpression":
+          return node.type === "TSAsExpression";
+
         case "CallExpression":
         case "NewExpression":
           return name === "callee" && parent.callee === node;

--- a/tests/typescript_as/as.js
+++ b/tests/typescript_as/as.js
@@ -20,3 +20,7 @@ const state = JSON.stringify({
 } as State);
 
 (foo.bar as Baz) = [bar];
+
+(bValue as boolean) ? 0 : -1;
+<boolean>bValue ? 0 : -1;
+

